site/docs/tasks/custom-taxonomy.md

```diff
@@ -51,3 +51,123 @@ helm install fybrik fybrik-charts/fybrik -n fybrik-system --wait --set-file taxo
+    required:
+      - intent
+```
+Copy the taxonomy layer to a `taxonomy-layer.yaml` file.
+
+The working directory is the fybrik repository.
+In order to compile and merge the two taxonomies, the Taxonomy Compile CLI tool can be used in the following way:
+
+```bash
+  go run main.go taxonomy compile --out custom-taxonomy.json --base charts/fybrik/files/taxonomy/taxonomy.json taxonomy-layer.yaml
+```
+
+This command will create a `custom-taxonomy.json` file, which will be used to install the fybrik helm chart using the following command:
+
+```bash
+helm install fybrik charts/fybrik --set global.tag=master --set global.imagePullPolicy=Always -n fybrik-system --wait --set-file taxonomyOverride=custom-taxonomy.json
+```
+
+Trying to deploy a fybrikapplication.yaml that has an intent of `Marketing` should fail validation beacuse there is no `Marketing` intent in the taxonomy. The following command should fail with a description of a validation error :
+
+```yaml
+cat << EOF | kubectl apply -f -
+apiVersion: app.fybrik.io/v1alpha1
+kind: FybrikApplication
+metadata:
+  name: taxonomy-test
+spec:
+  selector:
+   workloadSelector:
+     matchLabels: {
+       app: notebook
+     }
+  appInfo:
+    intent: Marketing
+    role: Business Analyst
+  data:
+    - dataSetID: "default/fake.csv"
+      requirements:
+        interface:
+          protocol: s3
+          dataformat: csv
+EOF
+```
+The expected error is `The FybrikApplication "taxonomy-test" is invalid: spec.appInfo.intent: Invalid value: "Marketing": spec.appInfo.intent must be one of the following: "Customer Behavior Analysis", "Customer Support", "Fraud Detection"`. Thus, no FybrikApplication CRD was created.
+
+To fix this, a new intent with `Marketing` value should be added to the taxonomy. Add a new value of "Marketing" in `custom-taxonomy.json` file in `intent` property to look like this:
+
+```
+"intent": {
+  "type": "string",
+  "enum": [
+    "Customer Behavior Analysis",
+    "Customer Support",
+    "Fraud Detection",
+    "Marketing"
+  ]
+}
+```
+
+Now, an upgrading to fybrik helm chart is needed using the following command:
+
+```bash
+helm upgrade fybrik charts/fybrik --set global.tag=master --set global.imagePullPolicy=Always -n fybrik-system --wait --set-file taxonomyOverride=custom-taxonomy.json
+```
+
+After updating fybrik to get fybrikapplications with `Marketing` intent, the deployment of a fybrikapplication.yaml that has an intent of `Marketing` should succeed:
+
+```yaml
+cat << EOF | kubectl apply -f -
+apiVersion: app.fybrik.io/v1alpha1
+kind: FybrikApplication
+metadata:
+  name: taxonomy-test
+spec:
+  selector:
+   workloadSelector:
+     matchLabels: {
+       app: notebook
+     }
+  appInfo:
+    intent: Marketing
+    role: Business Analyst
+  data:
+    - dataSetID: "default/fake.csv"
+      requirements:
+        interface:
+          protocol: s3
+          dataformat: csv
+EOF
+```
+
+It msut create a `taxonomy-test` FybrikApplication CRD.
```

Comment:
msut -> must

Reply:
The result is a FybrikApplication Custom Resource Definition instance called `taxonomy-test`